File: vacxcare-backend/src/models/SystemSettings.ts
import mongoose, { Schema, Document } from "mongoose";

export interface ISystemSettings extends Document {
  appName: string;
  appSubtitle?: string;
  logoUrl: string;
  primaryColor: string;
  headerColor?: string;
  headerTextColor?: string;
  headerIconColor?: string;
  sidebarBgColor?: string;
  sidebarTextColor?: string;
  accentColor?: string;
  language: string;
  timezone: string;
  // 📱 Paramètres mobile
  mobileBackgroundColor?: string;
  mobileButtonColor?: string;
  // Onboarding slides
  onboardingSlide1Image?: string;
  onboardingSlide1Title?: string;
  onboardingSlide1Subtitle?: string;
  onboardingSlide2Image?: string;
  onboardingSlide2Title?: string;
  onboardingSlide2Subtitle?: string;
  onboardingSlide3Image?: string;
  onboardingSlide3Title?: string;
  onboardingSlide3Subtitle?: string;
  // Dashboard header slides
  dashboardSlide1Image?: string;
  dashboardSlide1Title?: string;
  dashboardSlide1Subtitle?: string;
  dashboardSlide2Image?: string;
  dashboardSlide2Title?: string;
  dashboardSlide2Subtitle?: string;
  dashboardSlide3Image?: string;
  dashboardSlide3Title?: string;
  dashboardSlide3Subtitle?: string;
  // Autres paramètres
  notificationsEnabled: boolean;
  emailNotifications: boolean;
  smsNotifications: boolean;
  pushNotifications: boolean;
  notificationEmail?: string;
  notificationPhone?: string;
  maintenanceMode: boolean;
  maintenanceMessage?: string;
  backupEnabled: boolean;
  backupFrequency: string;
  sessionTimeout: number;
  maxLoginAttempts: number;
  passwordMinLength: number;
  requireSpecialChar: boolean;
  twoFactorEnabled: boolean;
  notificationChannels: {
    alerts: ("inapp" | "email" | "sms")[];
    auth: ("inapp" | "email" | "sms")[];
    onboarding_parent: ("inapp" | "email" | "sms")[];
  };
}

const SystemSettingsSchema = new Schema<ISystemSettings>(
  {
    appName: { type: String, default: "VacXCare" },
    appSubtitle: { type: String, default: "Santé de votre enfant simplifiée" },
    logoUrl: { type: String, default: "/logo.png" },
    primaryColor: { type: String, default: "#2563eb" },
    // 🎨 Thème
    headerColor: { type: String, default: "#ffffff" },
    headerTextColor: { type: String, default: "#1f2937" },
    headerIconColor: { type: String, default: "#6b7280" },
    sidebarBgColor: { type: String, default: "#0A1A33" },
    sidebarTextColor: { type: String, default: "#ffffff" },
    accentColor: { type: String, default: "#2563eb" },
    language: { type: String, default: "fr" },
    timezone: { type: String, default: "Africa/Dakar" },
    // 📱 Paramètres mobile
    mobileBackgroundColor: { type: String, default: "#0A1A33" },
    mobileButtonColor: { type: String, default: "#3B760F" },
    // Onboarding slides
    onboardingSlide1Image: { type: String },
    onboardingSlide1Title: { type: String, default: "Calendrier vaccinal simplifié" },
    onboardingSlide1Subtitle: { type: String, default: "Consultez tous les rendez-vous de vaccination de vos enfants en un seul endroit." },
    onboardingSlide2Image: { type: String },
    onboardingSlide2Title: { type: String, default: "Suivi professionnel et personnalisé" },
    onboardingSlide2Subtitle: { type: String, default: "Des agents de santé qualifiés pour accompagner chaque étape de la vaccination." },
    onboardingSlide3Image: { type: String },
    onboardingSlide3Title: { type: String, default: "Notifications et rappels intelligents" },
    onboardingSlide3Subtitle: { type: String, default: "Ne manquez plus jamais un vaccin important pour la santé de votre enfant." },
    // Dashboard header slides
    dashboardSlide1Image: { type: String },
    dashboardSlide1Title: { type: String, default: "Suivi Vaccinal Complet" },
    dashboardSlide1Subtitle: { type: String, default: "Tous les vaccins de votre enfant en un clin d'œil" },
    dashboardSlide2Image: { type: String },
    dashboardSlide2Title: { type: String, default: "Rendez-vous à Venir" },
    dashboardSlide2Subtitle: { type: String, default: "Ne manquez jamais un rendez-vous important" },
    dashboardSlide3Image: { type: String },
    dashboardSlide3Title: { type: String, default: "Santé de Votre Enfant" },
    dashboardSlide3Subtitle: { type: String, default: "Suivez la croissance et le développement" },
    // Autres paramètres
    notificationsEnabled: { type: Boolean, default: true },
    emailNotifications: { type: Boolean, default: true },
    smsNotifications: { type: Boolean, default: false },
    pushNotifications: { type: Boolean, default: false },
    notificationEmail: { type: String },
    notificationPhone: { type: String },
    maintenanceMode: { type: Boolean, default: false },
    maintenanceMessage: { type: String },
    backupEnabled: { type: Boolean, default: true },
    backupFrequency: { type: String, default: "daily" },
    sessionTimeout: { type: Number, default: 3600 },
    maxLoginAttempts: { type: Number, default: 5 },
    passwordMinLength: { type: Number, default: 8 },
    requireSpecialChar: { type: Boolean, default: true },
    twoFactorEnabled: { type: Boolean, default: false },
    notificationChannels: {
      type: new Schema(
        {
          alerts: { type: [String], default: ["inapp"] },
          auth: { type: [String], default: ["email"] },
          onboarding_parent: { type: [String], default: ["sms"] },
        },
        { _id: false }
      ),
      default: undefined,
    },
  },
  { timestamps: true }
);

export default mongoose.model<ISystemSettings>("SystemSettings", SystemSettingsSchema);

File: vacxcare_mobile/lib/models/system_settings.dart
class SystemSettings {
  final String appName;
  final String? appSubtitle;
  final String? logoUrl;
  final String? primaryColor;
  final String? mobileBackgroundColor;
  final String? mobileButtonColor;
  
  // Onboarding Slide 1
  final String? onboardingSlide1Image;
  final String? onboardingSlide1Title;
  final String? onboardingSlide1Subtitle;
  
  // Onboarding Slide 2
  final String? onboardingSlide2Image;
  final String? onboardingSlide2Title;
  final String? onboardingSlide2Subtitle;
  
  // Onboarding Slide 3
  final String? onboardingSlide3Image;
  final String? onboardingSlide3Title;
  final String? onboardingSlide3Subtitle;
  
  // Dashboard Slide 1
  final String? dashboardSlide1Image;
  final String? dashboardSlide1Title;
  final String? dashboardSlide1Subtitle;
  
  // Dashboard Slide 2
  final String? dashboardSlide2Image;
  final String? dashboardSlide2Title;
  final String? dashboardSlide2Subtitle;
  
  // Dashboard Slide 3
  final String? dashboardSlide3Image;
  final String? dashboardSlide3Title;
  final String? dashboardSlide3Subtitle;

  SystemSettings({
    required this.appName,
    this.appSubtitle,
    this.logoUrl,
    this.primaryColor,
    this.mobileBackgroundColor,
    this.mobileButtonColor,
    this.onboardingSlide1Image,
    this.onboardingSlide1Title,
    this.onboardingSlide1Subtitle,
    this.onboardingSlide2Image,
    this.onboardingSlide2Title,
    this.onboardingSlide2Subtitle,
    this.onboardingSlide3Image,
    this.onboardingSlide3Title,
    this.onboardingSlide3Subtitle,
    this.dashboardSlide1Image,
    this.dashboardSlide1Title,
    this.dashboardSlide1Subtitle,
    this.dashboardSlide2Image,
    this.dashboardSlide2Title,
    this.dashboardSlide2Subtitle,
    this.dashboardSlide3Image,
    this.dashboardSlide3Title,
    this.dashboardSlide3Subtitle,
  });

  factory SystemSettings.fromJson(Map<String, dynamic> json) {
    return SystemSettings(
      appName: json['appName'] ?? 'VaxCare',
      appSubtitle: json['appSubtitle'] ?? 'Santé de votre enfant simplifiée',
      logoUrl: json['logoUrl'],
      primaryColor: json['primaryColor'],
      mobileBackgroundColor: json['mobileBackgroundColor'] ?? '#0A1A33',
      mobileButtonColor: json['mobileButtonColor'] ?? '#3B760F',
      onboardingSlide1Image: json['onboardingSlide1Image'],
      onboardingSlide1Title: json['onboardingSlide1Title'] ?? 'Calendrier vaccinal simplifié',
      onboardingSlide1Subtitle: json['onboardingSlide1Subtitle'] ?? 'Consultez tous les rendez-vous de vaccination de vos enfants en un seul endroit.',
      onboardingSlide2Image: json['onboardingSlide2Image'],
      onboardingSlide2Title: json['onboardingSlide2Title'] ?? 'Suivi professionnel et personnalisé',
      onboardingSlide2Subtitle: json['onboardingSlide2Subtitle'] ?? 'Des agents de santé qualifiés pour accompagner chaque étape de la vaccination.',
      onboardingSlide3Image: json['onboardingSlide3Image'],
      onboardingSlide3Title: json['onboardingSlide3Title'] ?? 'Notifications et rappels intelligents',
      onboardingSlide3Subtitle: json['onboardingSlide3Subtitle'] ?? 'Ne manquez plus jamais un vaccin important pour la santé de votre enfant.',
      dashboardSlide1Image: json['dashboardSlide1Image'],
      dashboardSlide1Title: json['dashboardSlide1Title'] ?? 'Suivi Vaccinal Complet',
      dashboardSlide1Subtitle: json['dashboardSlide1Subtitle'] ?? 'Tous les vaccins de votre enfant en un clin d\'œil',
      dashboardSlide2Image: json['dashboardSlide2Image'],
      dashboardSlide2Title: json['dashboardSlide2Title'] ?? 'Rendez-vous à Venir',
      dashboardSlide2Subtitle: json['dashboardSlide2Subtitle'] ?? 'Ne manquez jamais un rendez-vous important',
      dashboardSlide3Image: json['dashboardSlide3Image'],
      dashboardSlide3Title: json['dashboardSlide3Title'] ?? 'Santé de Votre Enfant',
      dashboardSlide3Subtitle: json['dashboardSlide3Subtitle'] ?? 'Suivez la croissance et le développement',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'appName': appName,
      'appSubtitle': appSubtitle,
      'logoUrl': logoUrl,
      'primaryColor': primaryColor,
      'mobileBackgroundColor': mobileBackgroundColor,
      'mobileButtonColor': mobileButtonColor,
      'onboardingSlide1Image': onboardingSlide1Image,
      'onboardingSlide1Title': onboardingSlide1Title,
      'onboardingSlide1Subtitle': onboardingSlide1Subtitle,
      'onboardingSlide2Image': onboardingSlide2Image,
      'onboardingSlide2Title': onboardingSlide2Title,
      'onboardingSlide2Subtitle': onboardingSlide2Subtitle,
      'onboardingSlide3Image': onboardingSlide3Image,
      'onboardingSlide3Title': onboardingSlide3Title,
      'onboardingSlide3Subtitle': onboardingSlide3Subtitle,
      'dashboardSlide1Image': dashboardSlide1Image,
      'dashboardSlide1Title': dashboardSlide1Title,
      'dashboardSlide1Subtitle': dashboardSlide1Subtitle,
      'dashboardSlide2Image': dashboardSlide2Image,
      'dashboardSlide2Title': dashboardSlide2Title,
      'dashboardSlide2Subtitle': dashboardSlide2Subtitle,
      'dashboardSlide3Image': dashboardSlide3Image,
      'dashboardSlide3Title': dashboardSlide3Title,
      'dashboardSlide3Subtitle': dashboardSlide3Subtitle,
    };
  }
}
